src/xunit.netcore.extensions/Discoverers/ConditionalFactDiscoverer.cs

```diff
@@ -23,45 +24,71 @@ public ConditionalFactDiscoverer(IMessageSink diagnosticMessageSink) : base(diag
         public override IEnumerable<IXunitTestCase> Discover(
             ITestFrameworkDiscoveryOptions discoveryOptions, ITestMethod testMethod, IAttributeInfo factAttribute)
         {
-            MethodInfo testMethodInfo = testMethod.Method.ToRuntimeMethod();
+            string[] conditionMemberNames = factAttribute.GetConstructorArguments().FirstOrDefault() as string[];
+            IEnumerable<IXunitTestCase> testCases = base.Discover(discoveryOptions, testMethod, factAttribute);
+            return DiscoverConditionalTestCases(discoveryOptions, _diagnosticMessageSink, testMethod, testCases, conditionMemberNames);
+        }
 
-            string conditionMemberName = factAttribute.GetConstructorArguments().FirstOrDefault() as string;
-            Type declaringType = testMethodInfo.DeclaringType;
-            string[] symbols = conditionMemberName.Split('.');
+        // This helper method evaluates the given condition member names for a given set of test cases.
+        // If any condition member evaluates to 'false', the test cases are marked to be skipped.
+        // The skip reason is the collection of all the condition members that evalated to 'false'.
+        internal static IEnumerable<IXunitTestCase> DiscoverConditionalTestCases(
```

Comment:
Nit picks appreciated and all make sense.  I'll pull this helper stuff into a separate class, because it does feel odd for any Theory stuff to be using Fact methods.  I'll push up a new iteration shortly and wait for @Priya91 or @iamjasonp to comment.